bson/bsoncodec/empty_interface_codec.go

```diff
@@ -23,7 +23,7 @@ type EmptyInterfaceCodec struct {
 var (
 	defaultEmptyInterfaceCodec = NewEmptyInterfaceCodec()
 
-	_ ValueCodec  = defaultEmptyInterfaceCodec
+	_ valueCodec  = defaultEmptyInterfaceCodec
```

Comment:
This compile-time assertion is actually redundant because the codec is already passed into `RegisterTypeEncoder` and `RegisterTypeDecoder`, so I will remove it.